### PR TITLE
pagination in all courses

### PR DESCRIPTION
**Pull request by jatinkatyal13**

### CODE REVIEW

**Comment by abhishek97** [comment]
This pull request **fixes 1 alert** when merging 147348f3044d4680c4e036d4acb67c952f746048 into 6b0f82a5a1bb4262b48b3e927cb2699a19e41a6a - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/codingblocks.online.projectx/rev/pr-b68389b539a77d5a18d8e08e77a9eb31133a229f)

**fixed alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: app/pods/components/my-courses/component.js [hunk 1]
```diff
@@ -1,27 +1,37 @@
...
-import { readOnly } from '@ember-decorators/object/computed';
-
 
 export default class MyCoursesTaskComponent extends Component {
   @service store;
   @service api;
-  @readOnly("fetchMyCoursesTask.lastSuccessful.value") runs
+  runs = null
+
+  LIMIT = 8
```
Comment (abhishek97) [inline app/pods/components/my-courses/component.js]: Why you left caps lock ON?

File: app/pods/components/my-courses/template.hbs [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
   My Courses
 </div>
-<div class="row justify-content-center mb-5">
-  {{#if fetchMyCoursesTask.isRunning}}
+<div class="row justify-content-center">
+  {{#if (and (not runs.length) fetchMyCoursesTask.isRunning)}}
+  {{!-- {{#if fetchMyCoursesTask.isRunning}} --}}
     {{loading-card}}
   {{else}}
     {{#if runs.length}}
```
Comment (abhishek97) [inline app/pods/components/my-courses/template.hbs]: You can ditch the `if` block and move the `else` block to an `unless`. 
Also, try not to use more than one conditional helpers, this is an anti-pattern. Use aptly named computed properties instead.

File: app/pods/components/my-courses/component.js [hunk 3]
```diff
@@ -1,27 +1,37 @@
...
+        offset: this.OFFSET,
+        limit: this.LIMIT
+      }
     });
-  }
-
+    this.set('OFFSET', this.OFFSET + this.LIMIT)
     
+    if (!this.runs) {
+      this.set('runs', runs.toArray())
```
Comment (abhishek97) [inline app/pods/components/my-courses/component.js]: why is `.toArray()` necessary?

**Comment by ananay** [comment]
Thank you for your pull request. Pagination has been added to the All Courses. Closing.

**Pull request closed without merging**